Content.Server/Weapon/Melee/MeleeWeaponSystem.cs

```diff
@@ -43,8 +49,25 @@ public override void Initialize()
             SubscribeLocalEvent<MeleeWeaponComponent, ClickAttackEvent>(OnClickAttack);
             SubscribeLocalEvent<MeleeWeaponComponent, WideAttackEvent>(OnWideAttack);
             SubscribeLocalEvent<MeleeChemicalInjectorComponent, MeleeHitEvent>(OnChemicalInjectorHit);
+            SubscribeLocalEvent<MeleeChemicalInjectorComponent, GettingPickedUpAttemptEvent>(OnGettingPickedUpAttemptEvent);
+        }
+
+        private void OnGettingPickedUpAttemptEvent(EntityUid uid, MeleeChemicalInjectorComponent component, GettingPickedUpAttemptEvent args)
+        {
+            if (component.RequiredProtection == null || args.Cancelled)
+                return;
+            if (!_inventorySystem.TryGetSlotEntity(args.User, "gloves", out EntityUid? protection)
+                || !component.RequiredProtection.IsValid((EntityUid) protection))
+            {
+                //this is probably ugly, but I can't figure out a cleaner way to do it
+                List<EntityUid> thisList = new List<EntityUid>();
+                thisList.Add(args.User);
+                MeleeHitEvent thisEvent = new MeleeHitEvent(thisList, args.User);
+                TryInjectChemical(uid ,component ,thisEvent);
+            }
```

Comment:
Make the method take in a list instead of just allocating an event.

Reply:
done